script/publication_identifier_normalization.rb

```diff
@@ -0,0 +1,98 @@
+class PublicationIdentifierNormalization
+
+  attr_reader :logger
+
+  def initialize
+    @logger = Logger.new(Rails.root.join('log', 'publication_identifier_normalization.log'))
+  end
+
+  # Choose a parser that can handle the PublicationIdentifier.identifier_type
+  # @param pub_id [PublicationIdentifier]
+  # @return parser [ParseIdentifier] a kind of ParseIdentifier to handle the pub_id
+  def identifier_parser(pub_id)
+    case pub_id[:identifier_type]
```

Comment:
Using 
```ruby
case  pub_id[:identifier_type].downcase
```
would allow you to compare against strings instead of regexes.

Reply:
What does it matter?

Reply:
Actually, they're not the same and the regex is robust to `nil` values, i.e.
```ruby
[1] pry(main)> nil =~ /\Adoi\z/i
=> nil
[2] pry(main)> nil.downcase
NoMethodError: undefined method `downcase' for nil:NilClass
from (pry):2:in `__pry__'
```
This is a good case for the regex and leaving it as is.